src/hooks/useOnboard.ts

```diff
@@ -0,0 +1,125 @@
+import React from "react";
+import { Wallet } from "bnc-onboard/dist/src/interfaces";
+import Onboard from "bnc-onboard";
+import { Initialization } from "bnc-onboard/dist/src/interfaces";
+
+import { useConnection } from "./useConnection";
+
+const infuraId = process.env.REACT_APP_PUBLIC_INFURA_ID;
+
+const getNetworkName = (chainId: number) => {
+  switch (chainId) {
+    case 1: {
+      return "mainnet";
+    }
+    case 42: {
+      return "kovan";
+    }
+    case 3: {
+      return "ropsten";
+    }
+    case 4: {
+      return "rinkeby";
+    }
+  }
+};
+
+function onboardBaseConfig(_chainId?: number): Initialization {
+  const chainId = _chainId ?? 1;
+  const infuraRpc = `https://${getNetworkName(
+    chainId
+  )}.infura.io/v3/${infuraId}`;
+
+  return {
+    dappId: process.env.REACT_APP_ONBOARD_API_KEY || "",
+    hideBranding: true,
+    networkId: 1, // Default to main net. If on a different network will change with the subscription.
+    walletSelect: {
+      wallets: [
```

Comment:
can we just do the bare minimum wallet for first pass

Reply:
No problem!